hpOneView/image_streamer/resources/artifact_bundles.py

```diff
@@ -0,0 +1,375 @@
+        """
+        Updates only name for the Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            name_to_update: Artifact Bundle name to be updated.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
+                in OneView, just stop waiting for its completion.
+
+        Returns:
+            dict: Updated resource.
+        """
+        if self.URI not in id_or_uri:
+            id_or_uri = self.URI + "/" + id_or_uri
+
+        data = {
+            "type": "ArtifactsBundle",
+            "name": name_to_update,
+            "uri": id_or_uri
+        }
+
+        return self._client.update(data, timeout=timeout)
+
+    def extract_bundle(self, id_or_uri, timeout=-1):
+        """
+        Extracts the exisiting bundle on the appliance and creates all the artifacts.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation in
+                OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+        if self.URI not in id_or_uri:
+            id_or_uri = self.URI + "/" + id_or_uri
+
+        data = {
+            "uri": id_or_uri
+        }
+
+        custom_headers = {"Content-Type": "text/plain"}
+        return self._client.update(data, timeout=timeout, custom_headers=custom_headers)
+
+    def extract_backup_bundle(self, deployment_groups_id_or_uri, timeout=-1):
+        """
+        Extracts the exisiting backup bundle on the appliance and creates all the artifacts.
+
+        Args:
+            deployment_groups_id_or_uri: ID or URI of the Deployment Groups.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation in
+                OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+        if self.DEPLOYMENT_GROUPS_URI not in deployment_groups_id_or_uri:
+            deployment_groups_uri = self.DEPLOYMENT_GROUPS_URI + deployment_groups_id_or_uri
+        else:
+            deployment_groups_uri = deployment_groups_id_or_uri
+
+        data = {
+            "deploymentGroupURI": deployment_groups_uri
+        }
+
+        uri = self.URI + self.BACKUPS_PATH + self.ARCHIVE_PATH
+
+        return self._client.update(data, uri=uri, timeout=timeout)
+
+    def stop_artifact_creation(self, id_or_uri, task_uri):
+        """
+        Stops creation of the selected artifact bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            task_uri: Task URI associated with the Artifact Bundle.
+
+        Returns:
+            dict: The Artifact Bundle.
```

Comment:
Does it stops the remaining work of create the artifact but keep what was already ready? I mean, It stops creation but return a created bundle !?